jenkins/scripts/integration_test.sh

```diff
@@ -159,13 +167,19 @@ fi
 
 cat "${CI_DIR}/integration_test_env.sh" >> "${CI_DIR}/files/vars.sh"
 
+# Append to vars.sh in the remote vm 
```

Comment:
Why not write the whole file here first and then copy to the remote VM? Now we copy it over in two parts and have only half of it on the jenkins worker. This seems strange to me.

Reply:
If we write the file on the disk it will be overridden by the parallel test  which means will be moving wrong file content to one of the VMs, the part written on the disk is not conflicting but the other we write it directly on the target VM

Reply:
Ok, so this part is the same for both VMs? I think it would be better then to write to a random temp file. We could still get race conditions here even if the content is the same. For example, we could try to copy the file while the parallel job is still writing to it so we get half a file.

I would suggest to use `mktemp` to create a file instead of using `vars.sh`. Then we avoid the parallelism issues and can still write everything first before we copy it over. Something like `mktemp vars-XXXXXX.sh` should do the trick.

Reply:
done thanks